precompiles/erc20/query.go

```diff
@@ -20,3 +36,199 @@ const (
 	// query.
 	BalanceOfMethod = "balanceOf"
 )
+
+// Name returns the name of the token. If the token metadata is registered in the
+// bank module, it returns its name. Otherwise it returns the base denomination of
+// the token capitalized (eg. uatom -> Atom).
+func (p Precompile) Name(
+	ctx sdk.Context,
+	_ *vm.Contract,
+	_ vm.StateDB,
+	method *abi.Method,
+	_ []interface{},
```

Comment:
Why the unused params?

Reply:
there are not used in this query

Reply:
to answer your question, we've kept the same function for all queries, although we should revisit if this is a good pattern